### PR TITLE
test: Add tests to validate HelmReleases reference existing repos

### PR DESCRIPTION
**Pull request by cbuto**
This PR adds a test to validate that each `HelmRelease` in the `services` directory references a `HelmRepository` that is defined in the `helm-repositories` directory.

### CODE REVIEW

File: tests/test/services/services_test.go [hunk 1]
```diff
@@ -52,4 +54,30 @@ var _ = Describe("Services", func() {
...
+				if helmRelease.Spec.Chart.Spec.SourceRef.Kind == sourcev1.HelmRepositoryKind {
+					_, ok := helmRepos[helmRelease.Spec.Chart.Spec.SourceRef.Name]
+					// if ok is true, the helm repo referenced in the helm release exists
+					Expect(ok).To(BeTrue(),
+						fmt.Sprintf("the %s HelmRelease references a HelmRepository (%s) that doesn't exist in the %s directory",
+							helmRelease.Name, helmRelease.Spec.Chart.Spec.SourceRef.Name, HelmRepoDirectory))
+				}
+			})
+		}
+	})
```
[the remark this answers is not in the document]
Comment (cbuto) [reply tests/test/services/services_test.go:82]: I found that sometimes the subtests make it easier to figure out what failed.

For example, here is the test failure using the code above:
```
• [FAILED] [0.001 seconds]
Services
/Users/cbuto/Documents/projects/d2iq/dkp-catalog-applications/tests/test/services/services_test.go:20
  helmrepositories
  /Users/cbuto/Documents/projects/d2iq/dkp-catalog-applications/tests/test/services/services_test.go:57
    [It] each HelmRelease should reference a HelmRepository that exists in this repository
    /Users/cbuto/Documents/projects/d2iq/dkp-catalog-applications/tests/test/services/services_test.go:75

  Expected
      <map[string]string | len:3>: {
          "kubernetes-charts.banzaicloud.com": "${helmMirrorURL:=https://kubernetes-charts.banzaicloud.com}",
          "mesosphere.github.io-spark-charts": "${helmMirrorURL:=http://mesosphere.github.io/spark-on-k8s-operator}",
          "charts.pravega.io": "${helmMirrorURL:=https://charts.pravega.io}",
      }
  to contain elements
      <[]string | len:3, cap:3>: [
          "kubernetes-charts.banzaicloud.com",
          "mesosphere.github.io-spark-chartss",
          "charts.pravega.io",
      ]
  the missing elements were
      <[]string | len:3, cap:3>: [
          "kubernetes-charts.banzaicloud.com",
          "mesosphere.github.io-spark-chartss",
          "charts.pravega.io",
      ]
```

vs using subtests

```
Services
/Users/cbuto/Documents/projects/d2iq/dkp-catalog-applications/tests/test/services/services_test.go:21
  helmrepositories
  /Users/cbuto/Documents/projects/d2iq/dkp-catalog-applications/tests/test/services/services_test.go:58
    [It] the spark-operator HelmRelease should reference a HelmRepository that exists in this repository
    /Users/cbuto/Documents/projects/d2iq/dkp-catalog-applications/tests/test/services/services_test.go:72

  the spark-operator HelmRelease references a HelmRepository (mesosphere.github.io-spark-chartss) that doesn't exist in the ../../../helm-repositories directory
  Expected
      <map[string]string | len:3>: {
          "mesosphere.github.io-spark-charts": "${helmMirrorURL:=http://mesosphere.github.io/spark-on-k8s-operator}",
          "charts.pravega.io": "${helmMirrorURL:=https://charts.pravega.io}",
          "kubernetes-charts.banzaicloud.com": "${helmMirrorURL:=https://kubernetes-charts.banzaicloud.com}",
      }
  to have key
      <string>: mesosphere.github.io-spark-chartss
```

I'm good either way honestly, but that was the general idea of doing it this way

File: tests/pkg/files/files.go [hunk 2]
```diff
@@ -3,8 +3,15 @@ package files
 import (
 	"os"
 	"path"
+	"path/filepath"
+
+	v2 "github.com/fluxcd/helm-controller/api/v2beta1"
```
[the remark this answers is not in the document]
Comment (cbuto) [reply tests/pkg/files/files.go]: fixed in c4c48e18ab7c8d9de12bf8e04b0ef7ccb0fff3f3!

File: tests/test/services/services_test.go [hunk 3]
```diff
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"path"
 
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
```
[the remark this answers is not in the document]
Comment (cbuto) [reply tests/test/services/services_test.go]: fixed in c4c48e18ab7c8d9de12bf8e04b0ef7ccb0fff3f3!

File: tests/test/services/services_test.go [hunk 4]
```diff
@@ -52,4 +54,30 @@ var _ = Describe("Services", func() {
...
+		// Get list of all helm releases
+		helmReleases, err := files.ListHelmReleases(ServicesDirectory)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(len(helmReleases)).To(BeNumerically(">", 0))
+		// Validate each HelmRelease references an existing HelmRepository
+		for _, helmRelease := range helmReleases {
+			helmRelease := helmRelease
+			It(fmt.Sprintf("the %s HelmRelease should reference a HelmRepository that exists in this repository", helmRelease.Name), func() {
+				if helmRelease.Spec.Chart.Spec.SourceRef.Kind == sourcev1.HelmRepositoryKind {
```
[the remark this answers is not in the document]
Comment (cbuto) [reply tests/test/services/services_test.go]: i think you are right, i'm not sure if that would ever happen. fixed in c4c48e18ab7c8d9de12bf8e04b0ef7ccb0fff3f3!

File: tests/test/services/services_test.go [hunk 5]
```diff
@@ -52,4 +54,30 @@ var _ = Describe("Services", func() {
...
+					_, ok := helmRepos[helmRelease.Spec.Chart.Spec.SourceRef.Name]
+					// if ok is true, the helm repo referenced in the helm release exists
+					Expect(ok).To(BeTrue(),
```
[the remark this answers is not in the document]
Comment (cbuto) [reply tests/test/services/services_test.go]: definitely 😅, fixed in c4c48e18ab7c8d9de12bf8e04b0ef7ccb0fff3f3!

File: tests/test/services/services_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (takirala) [reply tests/test/services/services_test.go:80]: subtests approach sounds good, ty.

**Review by cwyl02 — APPROVED** [review]

**Pull request merged**